### PR TITLE
fix(store): check if link exists on write

### PR DESCRIPTION
**Pull request by walldiss**
In case of unexpected shutdown file might be partially written, while link is not. This PR will ensure we that put will result in both link and file on success.

File validation will be tackled in another PR

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -202,10 +202,6 @@ func (s *Store) createODSFile(
 ) (bool, error) {
 	pathODS := s.hashToPath(roots.Hash(), odsFileExt)
 	err := file.CreateODS(pathODS, roots, square)
-	if errors.Is(err, os.ErrExist) {
-		// TODO(@Wondertan): Should we verify that the exist file is correct?
-		return true, nil
-	}
 	if err != nil {
```
Comment (Wondertan) [inline store/store.go:205]:
```suggestion
	if err != nil && !errors.Is(err, os.ErrExist) {
```

**Review by Wondertan — COMMENTED** [review]
It asks for a test

File: store/store_test.go [hunk 2]
```diff
@@ -90,6 +90,30 @@ func TestEDSStore(t *testing.T) {
 		ensureAmountFileAndLinks(t, dir, 2, 1)
 	})
 
+	t.Run("Second Put after partial write", func(t *testing.T) {
+		dir := t.TempDir()
```
Comment (Wondertan) [inline store/store_test.go:94]: In case tests are run in parallel, do they have their own temp dir?

Comment (walldiss) [reply store/store_test.go:94]: yes

Comment (walldiss) [reply store/store_test.go:94]: Also test coverage is going to be greatly improved in following PR

**Review by Wondertan — APPROVED** [review]

**Comment by Wondertan** [comment]
For context: shwap-robus network has a few blocks missed in the storage, and the hypothesis is that they were missed during OOM leaving the Store in an inconsistent state(file exists, but link for it does not). This covers this possibility and we are yet to prove this is an issue

**Review by cristaloleg — APPROVED** [review]
SGTM

**Pull request merged**